operator/helper/bytesize_test.go

```diff
@@ -0,0 +1,119 @@
+package helper
+
+import (
+	"encoding/json"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	yaml "gopkg.in/yaml.v2"
+)
+
+func TestByteSizeUnmarshalJSON(t *testing.T) {
+	cases := []struct {
+		input       string
+		expected    ByteSize
+		expectError bool
+	}{
+		{`1`, 1, false},
+		{`1`, 1, false},
+		{`1`, 1, false},
+		{`3.3`, 3, false},
+		{`0`, 0, false},
+		{`10101010`, 10101010, false},
+		{`0.01`, 0, false},
+		{`"1"`, 1, false},
+		{`"1kb"`, 1000, false},
+		{`"1KB"`, 1000, false},
+		{`"1kib"`, 1024, false},
+		{`"1KiB"`, 1024, false},
+		{`"1mb"`, 1000 * 1000, false},
+		{`"1mib"`, 1024 * 1024, false},
+		{`"1gb"`, 1000 * 1000 * 1000, false},
+		{`"1gib"`, 1024 * 1024 * 1024, false},
+		{`"1tb"`, 1000 * 1000 * 1000 * 1000, false},
+		{`"1tib"`, 1024 * 1024 * 1024 * 1024, false},
+		{`"1pB"`, 1000 * 1000 * 1000 * 1000 * 1000, false},
+		{`"1pib"`, 1024 * 1024 * 1024 * 1024 * 1024, false},
+		{`1e3`, 1000, false},
+		{`"3ii3"`, 0, true},
+		{`3ii3`, 0, true},
+		{`--ii3`, 0, true},
+		{`{"test":"val"}`, 0, true},
+	}
+
+	for i, tc := range cases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			var bs ByteSize
+			err := json.Unmarshal([]byte(tc.input), &bs)
+			if tc.expectError {
+				require.Error(t, err)
+				return
+			}
+
+			require.NoError(t, err)
+			require.Equal(t, tc.expected, bs)
+		})
+	}
+}
+
+func TestByteSizeUnmarshalYAML(t *testing.T) {
+	cases := []struct {
+		input       string
+		expected    ByteSize
+		expectError bool
+	}{
+		{`1`, 1, false},
+		{`1`, 1, false},
+		{`3.3`, 3, false},
+		{`0`, 0, false},
+		{`10101010`, 10101010, false},
+		{`0.01`, 0, false},
+		{`"1"`, 1, false},
+		{`"1kb"`, 1000, false},
+		{`"1KB"`, 1000, false},
+		{`"1kib"`, 1024, false},
+		{`"1KiB"`, 1024, false},
+		{`"1mb"`, 1000 * 1000, false},
+		{`"1mib"`, 1024 * 1024, false},
+		{`"1gb"`, 1000 * 1000 * 1000, false},
+		{`"1gib"`, 1024 * 1024 * 1024, false},
+		{`"1tb"`, 1000 * 1000 * 1000 * 1000, false},
+		{`"1tib"`, 1024 * 1024 * 1024 * 1024, false},
+		{`"1pB"`, 1000 * 1000 * 1000 * 1000 * 1000, false},
+		{`"1pib"`, 1024 * 1024 * 1024 * 1024 * 1024, false},
+		{`1`, 1, false},
+		{`1kb`, 1000, false},
+		{`1KB`, 1000, false},
+		{`1kib`, 1024, false},
+		{`1KiB`, 1024, false},
+		{`1mb`, 1000 * 1000, false},
+		{`1mib`, 1024 * 1024, false},
+		{`1gb`, 1000 * 1000 * 1000, false},
+		{`1gib`, 1024 * 1024 * 1024, false},
+		{`1tb`, 1000 * 1000 * 1000 * 1000, false},
+		{`1tib`, 1024 * 1024 * 1024 * 1024, false},
+		{`1pB`, 1000 * 1000 * 1000 * 1000 * 1000, false},
+		{`1pib`, 1024 * 1024 * 1024 * 1024 * 1024, false},
```

Comment:
These cases appear to be duplicates.

At a higher level though, should we be able unmarshal the same values with either json or yaml? If so, it would probably make sense to share a single set of test inputs.

Reply:
They're not quite duplicates, they're just without quotes `""`. However, since YAML is a superset of JSON, all the JSON tests should work with YAML, so we can definitely reduce some duplication